torchrl/envs/custom/llm.py

```diff
@@ -4,23 +4,252 @@
 # LICENSE file in the root directory of this source tree.
 from __future__ import annotations
 
-from typing import Callable, List, Union
+from typing import Any, Callable, Literal
 
 import torch
-from tensordict import NestedKey, TensorDict, TensorDictBase
+from tensordict import NestedKey, TensorDict, TensorDictBase, unravel_key
 from tensordict.tensorclass import NonTensorData, NonTensorStack
-
+from tensordict.utils import _zip_strict
+from torch.utils.data import DataLoader
 from torchrl.data import (
+    Bounded,
     Categorical as CategoricalSpec,
     Composite,
     NonTensor,
     SipHash,
+    TensorSpec,
     Unbounded,
 )
 from torchrl.envs import EnvBase
 from torchrl.envs.utils import _StepMDP
 
 
+class LLMEnv(EnvBase):
+    """A text generation environment.
```

Comment:
Some verbiage on the "done" condition or this environment would be helpful

(iiuc user must append a transform, is that right?)